### ISSUE TITLE
Tito is not able to install package with not-yet-installed dependencies.

### ISSUE THREAD

**Issue by FrostyX** [dgoodwin/tito#233]
Tito is not able to `--install` package with not-yet-installed dependencies.
## Steps to reproduce

```
# Let us say that some new Requires was added to spec file
# For reproduce purposes, it is easier to uninstall some of required packages
$ sudo rpm -e python-simplejson --nodeps

# Now try to build and install
$ tito build --rpm --install
```
## Actual results

```
...

Successfully built: /tmp/tito/copr-cli-1.53-1.fc23.src.rpm /tmp/tito/noarch/copr-cli-1.53-1.fc23.noarch.rpm /tmp/tito/noarch/copr-cli-doc-1.53-1.fc23.noarch.rpm
Auto-installing packages:
sudo rpm -Uvh --force /tmp/tito/noarch/copr-cli-1.53-1.fc23.noarch.rpm /tmp/tito/noarch/copr-cli-doc-1.53-1.fc23.noarch.rpm
ERROR: Error running command: sudo rpm -Uvh --force /tmp/tito/noarch/copr-cli-1.53-1.fc23.noarch.rpm /tmp/tito/noarch/copr-cli-doc-1.53-1.fc23.noarch.rpm

Status code: 2

Command output: error: Failed dependencies:
    python-simplejson is needed by copr-cli-1.53-1.fc23.noarch

Traceback (most recent call last):
  File "/usr/bin/tito", line 23, in <module>
    CLI().main(sys.argv[1:])
  File "/usr/lib/python3.4/site-packages/tito/cli.py", line 202, in main
    return module.main(argv)
  File "/usr/lib/python3.4/site-packages/tito/cli.py", line 367, in main
    return builder.run(self.options)
  File "/usr/lib/python3.4/site-packages/tito/builder/main.py", line 148, in run
    self._auto_install()
  File "/usr/lib/python3.4/site-packages/tito/builder/main.py", line 302, in _auto_install
    run_command(cmd)
  File "/usr/lib/python3.4/site-packages/tito/common.py", line 427, in run_command
    raise RunCommandException(command, status, output)
tito.exception.RunCommandException: Error running command: sudo rpm -Uvh --force /tmp/tito/noarch/copr-cli-1.53-1.fc23.noarch.rpm /tmp/tito/noarch/copr-cli-doc-1.53-1.fc23.noarch.rpm
```
## Consequences

For instance in our Vagrantfile we have following workaround because of this

```
sudo rm -rf /tmp/tito/*
cd /vagrant/backend/ && tito build --test --rpm
sudo dnf -y install /tmp/tito/noarch/copr-backend*.noarch.rpm
```

instead of just simple

```
cd /vagrant/backend/ && tito build --test --rpm --install
```
## Suggested solution

I want to implement the solution and send PR, but first I want to discuss your ideas about it. 

Obviously we can't just use `dnf` isntead of `rpm` command because of systems that doesn't have it. However I would implement `--dnf` parameter which would be possible to chain with `--install` that would use `dnf` command and therefore install required dependencies.

Or even better, implement a mechanism to check whether `dnf` command is available and use it automatically. If not, fallback to `rpm`.

**Comment by dgoodwin** [comment]
Checking if dnf is available sounds great, if yum has an equivalent we could use that there as well, otherwise fall back to rpm. Preemptive thanks for the patch!

**Comment by FrostyX** [comment]
Thank you @dgoodwin for :+1: 
I am going to work on it this week and then I will be happy for some code review :-)

**Comment by jmrodri** [comment]
Fixed by PR #234
